### PR TITLE
Will's WIP branch

### PR DESCRIPTION
**Pull request by hellais**
I am opening a PR to make it easier to answer comments.

### CODE REVIEW

**Comment by hellais** [comment]
> Moved into the Default() method around instantiation of a default config:
> 90edffb
> Does that seem reasonable, or do you have a better suggestion?

My understanding of golang code style is that the `Default` method is usually meant to not have any consequences to calling it (it's actually used mostly as a kind of "property method").

I would suggest putting all the directory creation logic into some separate function (not a method) that is called at the beginning only if we detect that the system has not been initialised (or if the initialisation is partial).

**Comment by willscott** [comment]
refactored to a function as suggested in 62b515e2e0b6f9ed96284fed28300327ef604168

File: ooni.go [hunk 1]
```diff
@@ -140,11 +147,29 @@ func ParseConfig(b []byte) (*Config, error) {
...
+}
+
 // ReadConfig reads the configuration from the path
 func ReadConfig(path string) (*Config, error) {
 	b, err := ioutil.ReadFile(path)
 
 	if os.IsNotExist(err) {
+		if err = EnsureDefaultConfigDir(); err != nil {
+			return nil, errors.Wrap(err, "Creating new config")
+		}
```
Comment (hellais) [inline ooni.go]: I would not put this in here as this function is used to read the config file from a specific path. It doesn't make any assumption about what is on the disk. We should rather put this as something that is done when we construct the path to the config file inside of the ConfigDir

Comment (hellais) [reply ooni.go]: See: https://github.com/OpenObservatory/gooni/commit/efb7e87d1b66a7a5dad48f460a8f8cb553a3761c

**Pull request merged**